cql3/statements/call_statement.hh

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2020 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "cql3/cql_statement.hh"
+#include "cql3/column_identifier.hh"
+#include "cql3/term.hh"
+#include "cql3/stats.hh"
+#include "cql3/cf_name.hh"
+
+namespace db {
+    struct builtin_routine;
+}
+
+namespace cql3::statements {
+
+namespace raw { class call_statement; }
+
+/*
+ * CALL procedure();
+ *
+ * In case the procedure contains only DDL statements, returns no
+ * metadata. If a procedure returns a result set, returns result
+ * set metadata.
+ */
+class call_statement : public cql_statement_opt_metadata {
+private:
+    ::shared_ptr<cf_name> _cf_name;
```

Comment:
Nit: specifying `seastar` namespace is inconsistent.
Also, is there any polymorphism involved/planned here? If not, maybe it's possible to use `lw` pointers and make the `builtin_routine` `final`?